sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionContext.cs

```diff
@@ -29,21 +32,76 @@ public class PartitionContext
         /// <param name="consumerGroup">The name of the consumer group this context is associated with.</param>
         /// <param name="partitionId">The identifier of the Event Hub partition this context is associated with.</param>
+        /// <param name="partitionManager">Interacts with the storage system, dealing with the creation of checkpoints.</param>
+        /// <param name="ownerIdentifier">The identifier of the associated <see cref="EventProcessor{T}" /> instance.</param>
         ///
         internal PartitionContext(string eventHubName,
                                   string consumerGroup,
-                                  string partitionId)
+                                  string partitionId,
+                                  string ownerIdentifier,
+                                  PartitionManager partitionManager)
         {
             EventHubName = eventHubName;
             ConsumerGroup = consumerGroup;
             PartitionId = partitionId;
+            OwnerIdentifier = ownerIdentifier;
+            Manager = partitionManager;
+        }
+
+        /// <summary>
+        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        /// </summary>
+        ///
+        /// <param name="eventData">The event containing the information to be stored in the checkpoint.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public Task UpdateCheckpointAsync(EventData eventData) => UpdateCheckpointAsync(eventData.Offset.Value, eventData.SequenceNumber.Value);
+
+        /// <summary>
+        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        /// </summary>
+        ///
+        /// <param name="offset">The offset of the <see cref="EventData" /> the new checkpoint will be associated with.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the <see cref="EventData" /> the new checkpoint will be associated with.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public async Task UpdateCheckpointAsync(long offset,
+                                                long sequenceNumber)
+        {
+            // Parameter validation is done by Checkpoint constructor.
+
+            var checkpoint = new Checkpoint
+            (
+                EventHubName,
+                ConsumerGroup,
+                OwnerIdentifier,
+                PartitionId,
+                offset,
+                sequenceNumber
+            );
+
+            await Manager.UpdateCheckpointAsync(checkpoint).ConfigureAwait(false);
```

Comment:
I'd suggest just returning the task from `UpdateCheckpointAsync`;  you're not using it so there is no need to await.